src/App.tsx

```diff
@@ -1,26 +1,205 @@
+        setTodos(prevTodos => prevTodos.filter(todo => !todo.completed));
+      })
+      .catch(() => {
+        setErrorMessage(Errors.Delete);
+        wait(3000).then(() => setErrorMessage(null));
+      })
+      .finally(() => {
+        setDeletingIDs([]);
+      });
+  };
+
+  const visibleTodos = getFilteredTodos(todos, filterBy);
+
+  const activeTodosCount: number = todos.filter(todo => !todo.completed).length;
+  const hasCompletedTodo: boolean = todos.some(todo => todo.completed);
+
+  const handleSubmit = ({ title }: Todo) => {
+    if (!title.trim()) {
+      setErrorMessage(Errors.Empty);
+
+      return;
+    }
+
+    setIsDisabled(true);
+
+    setTempTodo({ id: 0, title, userId: USER_ID, completed: false });
+
+    createTodo({
+      title,
+      userId: USER_ID,
+      completed: false,
+    })
+      .then(newTodo => {
+        setTempTodo(null);
+        setTodos(prevTodos => [...prevTodos, newTodo]);
+        setTitleText('');
+      })
+      .catch(() => {
+        setErrorMessage(Errors.Add);
+        setTempTodo(null);
+      })
+      .finally(() => {
+        setIsDisabled(false);
+      });
+  };
+
+  const deleteTodo = (id: number) => {
+    setDeletingIDs(curIDs => [...curIDs, id]);
+
+    removeTodo(id)
+      .then(() => {
+        setTodos(prevTodos => prevTodos.filter(todo => todo.id !== id));
+      })
+      .catch(() => {
+        setErrorMessage(Errors.Delete);
+      })
+      .finally(() => {
+        setDeletingIDs([]);
+        inputRef.current?.focus();
+      });
+  };
+
+  useEffect(() => {
+    inputRef.current?.focus();
+    getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage(Errors.Load);
+      });
+  }, []);
+
+  useEffect(() => {
+    if (errorMessage) {
+      setTimeout(() => setErrorMessage(null), 3000);
+    }
+
+    inputRef.current?.focus();
+  }, [errorMessage, isDisabled]);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">
-          React Todo App - Load Todos
-        </a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          {/* this button should have `active` class only if all todos are completed */}
+          <button
+            type="button"
+            className="todoapp__toggle-all active"
+            data-cy="ToggleAllButton"
+          />
+
+          <form
+            onSubmit={() =>
+              handleSubmit({
+                title: titleText,
+                id: 0,
+                userId: 378,
+                completed: false,
+              })
```

Comment:
- You can get this object in your `handleSubmit` instead of passing it as an argument.
- And `378` is a magic value, create a constant for it or use if it's `USER_ID`